src/main.ts

```diff
@@ -0,0 +1,137 @@
+    try {
+        if (job_status.toUpperCase() === "CANCELLED") {
+            console.log("Job was cancelled, no notification will be sent")
+            process.exit(0);
+        }
+
+        let event_name : string = trigger_event_name || "event";
+        if (event_id) {
+            event_name = event_id;
+        }
+
+        switch(event_name){
+            case EventType.PUSH:
+                await schemaOnPush.validate(event_payload_text);
+                parsedSchema = schemaOnPush.cast(event_payload_text);
+                account = parsedSchema.pusher.name;
+                message = `**Commit to GitHub** by ${account}`;
+                url = parsedSchema.compare;
+                details = `Comment: ${parsedSchema.head_commit.message}`;
+                break;
+            case EventType.PULL_REQUEST:
+                await schemaOnPullRequest.validate(event_payload_text);
+                parsedSchema = schemaOnPullRequest.cast(event_payload_text);
+                account = parsedSchema.pull_request.user.login;
+                message =  `**PR submitted to Github** by ${account} with title: ${parsedSchema.pull_request.title}`;
+                url = parsedSchema.pull_request.html_url;
+                details = `Pr for merging ref ${parsedSchema.pull_request.head.ref} to base branch ${parsedSchema.pull_request.base.ref}`;
+                break;
+            case EventType.ISSUE:
+                await schemaOnIssue.validate(event_payload_text);
+                parsedSchema = schemaOnIssue.cast(event_payload_text);
+                account = parsedSchema.sender.login;
+                message = `**New/updated GitHub issue**: ${parsedSchema.issue.title}`;
+                url = parsedSchema.issue.html_url;
+                details = `Issue state: ${parsedSchema.issue.state}  - assignee: ${parsedSchema.issue.assignee}`;
+                break;
+            case EventType.ISSUE_COMMENT:
+                await schemaOnIssueComment.validate(event_payload_text);
+                parsedSchema = schemaOnIssueComment.cast(event_payload_text);
+                account = parsedSchema.sender.login;
+                message = `**A Github issue comment was posted** by ${account} with content ${parsedSchema.comment.body}`;
+                url = parsedSchema.issue.html_url;
+                details = `Issue state: ${parsedSchema.issue.state}  - assignee: ${parsedSchema.issue.assignee}`;
+                break;
+            default:
+                if (event_name) {
+                    message = event_name.replace(/_/g, ".");
+                } else {
+                    message = "A GitHub Actions event has occurred"
+                }
+                url = `https://github.com/${repository}/actions`;
+        }
+
+        if (!details) {
+            details = core.getInput('details');
+        }
+    } catch (error) {
+        core.setFailed(error.message);
+    }
+    return new NotificationMessage(message, url, details);
```

Comment:
I find it deeply fishy that we _always_ return a NotificationMessage even though we are performing validation that may fail and we have a try catch. What are we returning in the case of an error? Is it meaningful?

Reply:
this discussion has been taken to https://github.com/efcloud/el-msteams-post-action/pull/1#discussion_r386281947